### PR TITLE
[PT-BR] Translating JSDoc Reference, Configuring Watch, Project References and Declaration Merging

### PR DESCRIPTION
**Pull request by luk3skyw4lker**
Translating 3 files and fixing Creating DTS Files from JavaScript being on wrong directory.

### CODE REVIEW

**Comment by luk3skyw4lker** [comment]
@khaosdoctor @danilofuchs @orta 

I couldn't add you as reviewers (the option doesn't appear on my browser) so I'm mentioning you guys here. Thanks!

File: packages/documentation/copy/pt/project-config/Configuring Watch.md [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+
+## Contexto
+
+A implementação `--watch` do compilador se apoia em usar `fs.watch` e `fs.watchFile` que são providos pelo node, ambos os métodos têm seus prós e contras.
+
+`fs.watch` usa os eventos do file system para notificar as mudanças no arquivo/diretório. Mas isso depende do SO e a notificação não é completamente confiável e não funciona como esperado em vários SO. Também pode haver um limite de número de observadores que podem ser criados, eg. linux, e nós poderíamos superar esse limite bem rápido com programas que incluem grandes números de arquivos. Mas por isso usar os eventos do file system, não há tantos ciclos de CPU envolvidos. O compilador usa `fs.watch` tipicamente para observar diretórios (eg. diretórios fonte que são incluídos por um arquivo de configuração, diretório em que a resolução de módulos falhou etc.) estes podem lidar com a falta de precisão em notificação de mudanças. Mas observação recursiva é suportada apenas no Windows e no OSX. Isso significa que precisamos de algo para substituir a natureza recursiva em outros SO.
```
Comment (khaosdoctor) [inline packages/documentation/copy/pt/project-config/Configuring Watch.md]:
```suggestion
`fs.watch` usa os eventos do file system para notificar as mudanças no arquivo/diretório. Mas isso depende do SO e a notificação não é completamente confiável e não funciona como esperado em vários SOs. Também pode haver um limite do número de observadores que podem ser criados, eg. linux, e nós poderíamos superar esse limite bem rápido com programas que incluem grandes números de arquivos. Mas, por usar os eventos do file system, não há tantos ciclos de CPU envolvidos. O compilador usa `fs.watch` tipicamente para observar diretórios (eg. diretórios fonte que são incluídos por um arquivo de configuração, diretório em que a resolução de módulos falhou etc.) estes podem lidar com a falta de precisão em notificação de mudanças. Mas observação recursiva é suportada apenas no Windows e no OSX. Isso significa que precisamos de algo para substituir a natureza recursiva em outros SOs.
```

File: packages/documentation/copy/pt/project-config/Configuring Watch.md [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+    "watchDirectory": "useFsEvents",
+
+    // Checka os arquivos buscando atualizações mais frequentemente
+    // quando eles são muito atualizados
+    "fallbackPolling": "dynamicPriority"
+  }
+}
+```
+
+Você pode ler mais sobre isso nas [notas de lançamento](/docs/handbook/release-notes/typescript-3-8.html#better-directory-watching-on-linux-and-watchoptions).
```
Comment (khaosdoctor) [inline packages/documentation/copy/pt/project-config/Configuring Watch.md]:
```suggestion
Você pode ler mais sobre isso nas [notas de atualização](/docs/handbook/release-notes/typescript-3-8.html#better-directory-watching-on-linux-and-watchoptions).
```

File: packages/documentation/copy/pt/project-config/Configuring Watch.md [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+
+## Configurando observação de arquivos usando variáveis de ambiente `TSC_WATCHFILE`
+
+<!-- prettier-ignore -->
+Opção                                          | Descrição
+-----------------------------------------------|----------------------------------------------------------------------
+`PriorityPollingInterval`                      | Usa `fs.watchFile` mas usa diferentes intervalos de checagem para arquivos fonte, arquivos de configuração e arquivos em falta
+`DynamicPriorityPolling`                       | Usa uma fila dinâmica onde os arquivos modificados mais frequentemente serão checados em menor intervalo e os que se mantem inalterados serão checados em um intervalo maior
+`UseFsEvents`                                  | Usa o `fs.watch` que por sua vez usa os eventos de file system (mas pode não ser exato em diferentes SO) para conseguir as notificações de mudança/criação/deleção de arquivos. Note que alguns SO eg. linux tem um limite no número de observadores e a falha de criação de um observador com `fs.watch` resultará na criação de outro observador usando `fs.watchFile`
```
Comment (khaosdoctor) [inline packages/documentation/copy/pt/project-config/Configuring Watch.md]:
```suggestion
`UseFsEvents`                                  | Usa o `fs.watch` que por sua vez usa os eventos de file system (mas pode não ser exato em diferentes SOs) para conseguir as notificações de mudança/criação/remoção de arquivos. Note que alguns SOs eg. linux tem um limite no número de observadores e a falha de criação de um observador com `fs.watch` resultará na criação de outro observador usando `fs.watchFile`
```

File: packages/documentation/copy/pt/project-config/Configuring Watch.md [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+`UseFsEventsOnParentDirectory`                 | Essa opção observa o diretório pai do arquivo com `fs.watch` (usando eventos do file system) causando com que o uso de CPU seja baixo
+default (nenhum valor especificado)            | Se a variável de ambiente `TSC_NONPOLLING_WATCHER` tem valor true, observa os diretórios pai dos arquivos (assim como `UseFsEventsOnParentDirectory`). De outra forma observa os arquivos usando `fs.watchFile` com `250ms` como tempo limite para qualquer arquivo.
+
+## Configurando observação de diretório usando a variável de ambiente `TSC_WATCHDIRECTORY`
+
+A observação de diretório em plataformas que não suportam observação recursiva de diretório nativamente em node, é suportada por meio da criação recursiva de observadores de diretório para os diretórios filhos usando diferentes opções selecionadas pela variável ambiente `TSC_WATCHDIRECTORY`. Note que em plataformas que suportam observação de diretórios recursivamente (e.g. windows) o valor dessa variável é ignorado.
```
Comment (khaosdoctor) [inline packages/documentation/copy/pt/project-config/Configuring Watch.md]:
```suggestion
A observação de diretório em plataformas que não suportam observação recursiva de diretório nativamente em node é suportada por meio da criação recursiva de observadores de diretório para os diretórios filhos usando diferentes opções selecionadas pela variável ambiente `TSC_WATCHDIRECTORY`. Note que, em plataformas que suportam observação de diretórios recursivamente (e.g. windows), o valor dessa variável é ignorado.
```

File: packages/documentation/copy/pt/project-config/Project References.md [hunk 5]
```diff
@@ -0,0 +1,219 @@
...
+title: Referência de Projeto
+layout: docs
+permalink: /pt/docs/handbook/project-references.html
+oneline: Como dividir um projeto Typescript grande
+translatable: true
+---
+
+Referências de projeto são uma nova funcionalidade no Typescript 3.0 que permitem que você estruture seus programas Typescript em peças menores.
+
+Fazendo isso, você pode melhorar muito tempos de compilação, garantir serparação lógica entre componentes e organizar seu código de melhores maneiras.
```
Comment (khaosdoctor) [inline packages/documentation/copy/pt/project-config/Project References.md]:
```suggestion
Fazendo isso, você pode melhorar muito tempos de compilação, garantir separação lógica entre componentes e organizar seu código de melhores maneiras.
```

File: packages/documentation/copy/pt/project-config/Project References.md [hunk 6]
```diff
@@ -0,0 +1,219 @@
...
+
+- Era possível que os arquivos de implementação importassem os arquivos de teste
+- Não era possível compilar `test` e `src` ao mesmo tempo sem que `src` aparecesse na pasta de saída, o que você provavelmente não gostaria que acontecesse
+- Mudar apenas os arquivos de implementação exigiam que os testes passassem novamente por _checagem de tipo_ mesmo que isso não causasse novos erros
+- Mudar apenas os testes exigia a checagem de tipo novamente na implementação, mesmo que nada tivesse mudado
+
+Você poderia usar múltiplos arquivos tsconfig para resolver _alguns_ desses problemas, mas novos apareceriam:
+
+- Não há checagem atualizada embutida, então você acaba sempre rodando `tsc` duas vezes
+- Invoking `tsc` twice incurs more startup time overhead
```
Comment (khaosdoctor) [inline packages/documentation/copy/pt/project-config/Project References.md]:
```suggestion

```

File: packages/documentation/copy/pt/project-config/Project References.md [hunk 7]
```diff
@@ -0,0 +1,219 @@
...
+- Invocar `tsc` duas vezes implicaria em mais sobrecarga de tempo de início
```
Comment (khaosdoctor) [inline packages/documentation/copy/pt/project-config/Project References.md]:
```suggestion
- Invocar `tsc` duas vezes implicaria em mais sobrecarga no tempo de inicialização
```

File: packages/documentation/copy/pt/project-config/Project References.md [hunk 8]
```diff
@@ -0,0 +1,219 @@
...
+- Se o projeto referenciado produz um `outFile`, o arquivo de saída `.d.ts` daquele arquivo será visível neste projeto
+- Modo de compilação (veja abaixo) vai automaticamente compilar o projeto referenciado se preciso
+
+Por meio da separação em múltiplos projetos, você pode melhorar muito a velocidade de checagem de tipo e compilação, reduzir uso de memória usando um editor e melhorar a garantia de grupos lógicos de seu programa.
+
+## `composite`
+
+Projetos referenciados devem ter a nova configuração `composite` habilitada.
+Essa configuração é necessária para garantir que o Typescript possa rapidamente determinar onde encontrar as saídas do projeto referenciado.
+Enabling the `composite` flag changes a few things:
```
Comment (khaosdoctor) [inline packages/documentation/copy/pt/project-config/Project References.md]:
```suggestion

```

File: packages/documentation/copy/pt/project-config/Project References.md [hunk 9]
```diff
@@ -0,0 +1,219 @@
...
+Habilitar a configuração `composite` muda algumas coisas:
+
+- A configuração `rootDir`, se não explicitamente declarada, tem como padrão o diretório onde o arquivo `tsconfig` está
+- Todos os arquivos de implementação tem que ser cobridos por um padrão `include` ou listados no vetor `files`. Se essa restrição for violada, o `tsc` vai te informar de quais arquivos não foram especificados.
+- `declaration` tem que estar habilitada
+
+## `declarationMap`s
```
Comment (khaosdoctor) [inline packages/documentation/copy/pt/project-config/Project References.md]:
```suggestion
## `declarationMap`
```

File: packages/documentation/copy/pt/project-config/Project References.md [hunk 10]
```diff
@@ -0,0 +1,219 @@
...
+
+Nós também adicionamos suporte para [mapas de declaração de fontes](https://github.com/Microsoft/TypeScript/issues/14479).
+Se você habilitar `--declarationMap`, você será capaz de usar funcionalidades de editores tais como "Ir para definição" e Renomear para transparentemente navegar e editar código entre limites de projeto em editores que são suportados.
```
Comment (khaosdoctor) [inline packages/documentation/copy/pt/project-config/Project References.md]:
```suggestion
Se você habilitar `--declarationMap`, você será capaz de usar funcionalidades de editores tais como "Ir para definição" e Renomear para transparentemente navegar e editar código entre projetos em editores que são suportados.
```

File: packages/documentation/copy/pt/project-config/Project References.md [hunk 11]
```diff
@@ -0,0 +1,219 @@
...
+
+## `prepend` com `outFile`
+
+Você também pode habilitar a pre-adição de saídas de uma dependência usando a opção `prepend` em uma referência:
```
Comment (khaosdoctor) [inline packages/documentation/copy/pt/project-config/Project References.md]:
```suggestion
Você também pode habilitar a prefixação de saídas de uma dependência usando a opção `prepend` em uma referência:
```

File: packages/documentation/copy/pt/project-config/Project References.md [hunk 12]
```diff
@@ -0,0 +1,219 @@
...
+
+```js
+   "references": [
+       { "path": "../utils", "prepend": true }
+   ]
+```
+
+Pré-adicionar um projeto vai incluir a saída dele acima da saída do projeto atual.
```
Comment (khaosdoctor) [inline packages/documentation/copy/pt/project-config/Project References.md]:
```suggestion
Prefixar um projeto vai incluir a saída dele acima da saída do projeto atual.
```

File: packages/documentation/copy/pt/project-config/Project References.md [hunk 13]
```diff
@@ -0,0 +1,219 @@
...
+   A
+  ^ ^
+ /   \
+B     C
+ ^   ^
+  \ /
+   D
+```
+
+É importante nessa situação que não seja realizada a pré-adição em cada referência, porque você vai acabar tendo duas cópias de `A` na saída de `D` - isso pode levar a resultados inesperados.
```
Comment (khaosdoctor) [inline packages/documentation/copy/pt/project-config/Project References.md]:
```suggestion
É importante nessa situação que não seja realizada a prefixação em cada referência, porque você vai acabar tendo duas cópias de `A` na saída de `D` - isso pode levar a resultados inesperados.
```

File: packages/documentation/copy/pt/project-config/Project References.md [hunk 14]
```diff
@@ -0,0 +1,219 @@
...
+- `--force`: Age como se todos os projetos estivessem desatualizados
+- `--watch`: Mobo de observação (não pode ser combinado com qualquer opção exceto `--verbose`)
+
+## Ressalvas
+
+Normalmente `tsc` produzirá saídas (`.js` e `.d.ts`) na presença de erros de sintaxe ou de tipo, exceto que `noEmitOnError` esteja habilitada.
+Fazer isso em um sistema de compilação incremental seria bem ruim - se um dos seus projetos desatualizados tivesse um novo erro, você só veria ele _uma vez_ pois em uma compilação subsequente, a compilação deste projeto seria pulada, pois o projeto agora está atualizado.
+Por essa razão, `tsc -b` age efetivamente como se `noEmitOnError` estivesse habilitada para todos os projetos.
+
+If you check in any build outputs (`.js`, `.d.ts`, `.d.ts.map`, etc.), you may need to run a `--force` build after certain source control operations depending on whether your source control tool preserves timestamps between the local copy and the remote copy.
```
Comment (khaosdoctor) [inline packages/documentation/copy/pt/project-config/Project References.md]: Missing translation here

File: packages/documentation/copy/pt/project-config/Project References.md [hunk 15]
```diff
@@ -0,0 +1,219 @@
...
+
+## MSBuild
+
+Se você tem um projeto msbuild, você pode habilitar o modo de compilação adicionando
+
+```xml
+    <TypeScriptBuildMode>true</TypeScriptBuildMode>
+```
+
+ao seu arquivo proj. Isso vai habilitar compilação incremental e limpeza automáticas.
```
Comment (khaosdoctor) [inline packages/documentation/copy/pt/project-config/Project References.md]:
```suggestion
Ao seu arquivo proj. Isso vai habilitar compilação incremental e limpeza automáticas.
```

File: packages/documentation/copy/pt/reference/Declaration Merging.md [hunk 16]
```diff
@@ -0,0 +1,332 @@
...
+title: Fusão de Declarações
+layout: docs
+permalink: /pt/docs/handbook/declaration-merging.html
+oneline: Como a fusão de namespaces e interfaces funciona
+translatable: true
+---
+
+## Introdução
+
+Alguns dos conceitos únicos em Typescript descreve a forma de um objeto Javascript ao nível de tipo.
```
Comment (khaosdoctor) [inline packages/documentation/copy/pt/reference/Declaration Merging.md]:
```suggestion
Alguns dos conceitos únicos em Typescript descrevem a forma de um objeto Javascript ao nível de tipo.
```

File: packages/documentation/copy/pt/reference/Declaration Merging.md [hunk 17]
```diff
@@ -0,0 +1,332 @@
...
+Um exemplo que é especialmente único do Typescript é o coneito de 'fusão de declarações'.
```
Comment (khaosdoctor) [inline packages/documentation/copy/pt/reference/Declaration Merging.md]:
```suggestion
Um exemplo que é especialmente único do Typescript é o conceito de 'fusão de declarações'.
```

File: packages/documentation/copy/pt/reference/Declaration Merging.md [hunk 18]
```diff
@@ -0,0 +1,332 @@
...
+  createElement(tagName: "div"): HTMLDivElement;
+  createElement(tagName: "span"): HTMLSpanElement;
+  createElement(tagName: string): HTMLElement;
+  createElement(tagName: any): Element;
+}
+```
+
+## Fundindo Namespaces
+
+Similar às interfaces, namespaces de mesmo nome serão também fundir seus membros.
```
Comment (khaosdoctor) [inline packages/documentation/copy/pt/reference/Declaration Merging.md]:
```suggestion
Similar às interfaces, namespaces de mesmo nome terão seus membros fundidos.
```

File: packages/documentation/copy/pt/reference/Declaration Merging.md [hunk 19]
```diff
@@ -0,0 +1,332 @@
...
+  let haveMuscles = true;
+
+  export function animalsHaveMuscles() {
+    return haveMuscles;
+  }
+}
+
+namespace Animal {
+  export function doAnimalsHaveMuscles() {
+    return haveMuscles; // Error, because haveMuscles is not accessible here
```
Comment (khaosdoctor) [inline packages/documentation/copy/pt/reference/Declaration Merging.md]:
```suggestion
    return haveMuscles; // Erro, porque haveMuscles não é acessível aqui
```

File: packages/documentation/copy/pt/reference/Declaration Merging.md [hunk 20]
```diff
@@ -0,0 +1,332 @@
...
+  }
+}
+```
+
+Por `haveMuscles` não ser exportado, apenas a função `animalsHaveMuscles` que compartilha o mesmo namespace não fundido pode ver o símbolo.
+A função `doAnimalsHaveMuscles`, mesmo que seja parte do namespace fundido `Animal` não pode ver este membro não exportado.
+
+## Merging Namespaces with Classes, Functions, and Enums
```
Comment (khaosdoctor) [inline packages/documentation/copy/pt/reference/Declaration Merging.md]:
```suggestion

```

File: packages/documentation/copy/pt/reference/Declaration Merging.md [hunk 21]
```diff
@@ -0,0 +1,332 @@
...
+## Fusões Desencorajadas
+
+Nem todas as fusões são permitidas no TypeScript.
+Atualmente, classes não podem se fundir com outras classes ou variáveis.
+Para mais informações em simular a fusão de classes, veja a seção de [Mixins em TypeScript](/docs/handbook/mixins.html)
+
+## Aumento de Módulos
+
+Ainda que módulos JavaScript não suportem fusões, você pode alterar objetos existentes por meio da importação e atualização deles.
+Olhemos para um exemplo de brinquedo de Observable:
```
Comment (khaosdoctor) [inline packages/documentation/copy/pt/reference/Declaration Merging.md]:
```suggestion
Olhemos para um exemplo de Observable:
```

**Review by khaosdoctor — CHANGES_REQUESTED** [review]

**Comment by luk3skyw4lker** [comment]
Sorry for the dummy errors, it was my first time contributing in an open source project and it was long since I've last translated something from English into Portuguese hahaha

**Comment by khaosdoctor** [comment]
No worries! I'll merge it and verify using local editors :)

**Comment by khaosdoctor** [comment]
lgtm

**Pull request merged**